### PR TITLE
Refactor postgresql migration

### PR DESCRIPTION
**Pull request by ssh24**
### Description
Refactor discovery test cases

#### Related issues
connect to https://github.com/strongloop/loopback-connector/issues/15

### CODE REVIEW

File: lib/migration.js [hunk 1]
```diff
@@ -11,136 +11,43 @@ var async = require('async');
...
-      }
-    }
-
-    function decoratedTableDataCallback(err, data) {
-      if (err) {
-        console.error(err);
-      }
-      if (!err) {
-        data.forEach(function(field) {
-          field.type = mapPostgreSQLDatatypes(field.type, field.length);
```
Comment (jannyHou) [inline lib/migration.js]: `mapPostgreSQLDatatypes` is missing after refactor

File: lib/migration.js [hunk 2]
```diff
@@ -517,29 +378,9 @@ function mixinMigration(PostgreSQL) {
...
   };
 
-  /*!
-   * Find the column type for a given model property
-   *
-   * @param {String} model The model name
-   * @param {String} property The property name
-   * @returns {String} The column type
-   */
-  PostgreSQL.prototype.columnDataType = function(model, property) {
```
Comment (jannyHou) [inline lib/migration.js]: Hmm, why changing this function from `columnDataType` to `buildColumnType`?
we need both of them.

Comment (ssh24) [inline lib/migration.js]: Moved `columnDataType` to base connector https://github.com/strongloop/loopback-connector/pull/78/commits/752359febbc39abb5ecc4c38cb15ac18012c1112 
We still have `buildColumnType` in postgresql

**Comment by ssh24** [comment]
@slnode test please

**Pull request merged**